fiboa_cli/datasets/template.py

```diff
@@ -4,170 +4,143 @@
+    # This is useful if the data is split into multiple files/layers and columns should be added or changed
+    # on a per-file/layer basis for example.
+    # The path contains the local path to the file that was read.
+    # The uri contains the URL that was read.
+    # The layer may contain the layer name.
+    # def file_migration(self, gdf: gpd.GeoDataFrame, path: str, uri: str, layer: str = None) -> gpd.GeoDataFrame:
+    #     return data
+
+    # Schemas for the fields that are not defined in fiboa
+    # Keys must be the values from the COLUMNS dict, not the keys
+    missing_schemas = {
+        "required": ["my_id"], # i.e. non-nullable properties
+        "properties": {
+            "my_id": {
+                "type": "string"
+            }
+        }
+    }
 
-# Conversion function, usually no changes required
-def convert(output_file, cache = None, **kwargs):
     """
+    def convert(self, *args, **kwargs):
     Converts the field boundary datasets to fiboa.
 
     For reference, this is the order in which the conversion steps are applied:
-    0. Read GeoDataFrame from file(s) / layer(s) and run the FILE_MIGRATION function if provided
-    1. Run global migration (if provided through MIGRATION)
+    0. Read GeoDataFrame from file(s) / layer(s) and run self.file_migration()
+    1. Run global migration self.migrate()
     2. Run filters to remove rows that shall not be in the final data
-       (if provided through COLUMN_FILTERS)
+       (if provided through self.column_filters)
     3. Add columns with constant values
-    4. Run column migrations (if provided through COLUMN_MIGRATIONS)
-    5. Duplicate columns (if an array is provided as the value in COLUMNS)
-    6. Rename columns (as provided in COLUMNS)
-    7. Remove columns (if column is not present as value in COLUMNS)
+    4. Run column migrations (if provided through self.column_migrations)
+    5. Duplicate columns (if an array is provided as the value in self.columns)
+    6. Rename columns (as provided in self.columns)
+    7. Remove columns (if column is not present as value in self.columns)
     8. Create the collection
     9. Change data types of the columns based on the provided schemas
-    (fiboa spec, extensions, and MISSING_SCHEMAS)
+    (fiboa spec, extensions, and self.missing_schemas)
     10. Write the data to the Parquet file
-
-    Parameters:
-    output_file (str): Path where the Parquet file shall be stored.
-    cache (str): Path to a cached folder for the data. Default: None.
-                      Can be used to avoid repetitive downloads from the original data source.
-    source_coop_url (str): URL to the (future) Source Cooperative repository. Default: None
-    collection (bool): Additionally, store the collection separate from Parquet file. Default: False
-    compression (str): Compression method for the Parquet file. Default: brotli
-    kwargs: Additional keyword arguments for GeoPanda's read_file() or read_parquet() function.
```

Comment:
We should probably preserve the documentation somewhere if not done yet...

Reply:
Ah, we could keep the original template.py file as `template_convert_function.py` and remove it once everything is refactored... Done.

Reply:
That's not what I meant, but is also okay. :-)